src/main/java/domain/ProfitResult.java

```diff
@@ -0,0 +1,62 @@
+package domain;
+
+import domain.gambler.Dealer;
+import domain.gambler.Money;
+import domain.gambler.Player;
+import domain.gambler.Players;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ProfitResult {
+
+    private static final String NULL_EXCEPTION_MESSAGE = "Null exception.";
+
+    private final Map<Player, Money> playerResults;
+    private final Money dealerResults;
+
+    public ProfitResult(Dealer dealer, Players players) {
+        validateDealer(dealer);
+        validatePlayers(players);
+        this.playerResults = Collections.unmodifiableMap(calculatePlayerResults(dealer, players));
+        this.dealerResults = calculateDealerResults();
+    }
+
+    private void validatePlayers(Players players) {
+        if (Objects.isNull(players)) {
+            throw new IllegalArgumentException(NULL_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private void validateDealer(Dealer dealer) {
+        if (Objects.isNull(dealer)) {
+            throw new IllegalArgumentException(NULL_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private Map<Player, Money> calculatePlayerResults(Dealer dealer, Players players) {
+        return players.getPlayers().stream()
+                .collect(Collectors
+                        .toMap(player -> player, player -> player.getProfitByComparing(dealer),
+                                (a1, a2) -> a1, LinkedHashMap::new));
+    }
+
+    private Money calculateDealerResults() {
+        Money dealerMoney = Money.fromPositive("0");
```

Comment:
```suggestion
        Money dealerMoney = Money.Zero;
```

와 같이 기본 값을 만든다면 매직넘버를 제거한 효과를 누릴 수 있습니다!

Reply:
수정하였습니다 :)
혹시 enum에서도 매직넘버를 제거해주는 것이 좋은 방식인지 궁금합니다.
WinningResult의 
`WIN(WinningResult::isWin, profitRatio : 1)'을
`WIN(WinningResult::isWin, Constants.WIN_PROFIT_RATIO`
이런 식으로 바꿔주는 것이 이점이 있는지 궁금합니다.

Reply:
enum의 경우 일종의 구조체이기 때문에 따로 분리하지 않아도 괜찮습니다!